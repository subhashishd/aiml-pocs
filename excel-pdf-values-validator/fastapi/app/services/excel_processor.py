import logging
from typing import List, Dict, Any
import openpyxl
import pandas as pd

logger = logging.getLogger(__name__)

class ExcelProcessor:
    def process_excel(self, file_path: str) -> List[Dict[str, Any]]:
        """
        Process Excel file to extract parameter-value pairs
        Expected format: Parameter | Value | Unit
        """
        try:
            logger.info(f"Processing Excel file: {file_path}")
            
            # Read Excel file
            df = pd.read_excel(file_path)
            
            # Ensure required columns exist
            expected_columns = ['Parameter', 'Value', 'Unit']
            missing_columns = [col for col in expected_columns if col not in df.columns]
            
            if missing_columns:
                logger.warning(f"Missing columns: {missing_columns}")
                # Try to find columns with similar names
                df.columns = df.columns.str.strip().str.title()
            
            data = []
            for _, row in df.iterrows():
                if pd.notna(row.get('Parameter')) and pd.notna(row.get('Value')):
                    parameter = str(row['Parameter']).strip()
                    value = row['Value']
                    unit = str(row.get('Unit', '')).strip()
                    
                    # Convert value to appropriate type
                    if isinstance(value, str):
                        try:
                            # Try to convert to float for numeric values
                            value = float(value)
                        except ValueError:
                            pass  # Keep as string if not numeric
                    
                    data.append({
                        'parameter': parameter,
                        'value': value,
                        'unit': unit,
                        'original_text': f"{parameter}: {value} {unit}".strip()
                    })
            
            logger.info(f"Extracted {len(data)} parameter-value pairs from Excel")
            return data
            
        except Exception as e:
            logger.error(f"Error processing Excel file: {str(e)}")
            raise
