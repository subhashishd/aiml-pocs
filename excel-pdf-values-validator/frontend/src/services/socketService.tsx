import React, { createContext, useContext, useEffect, useState, ReactNode } from 'react';
import { io, Socket } from 'socket.io-client';
import toast from 'react-hot-toast';

// Socket events interface
interface SocketEvents {
  // Task events
  'task:created': (data: { taskId: string; filename: string; status: string }) => void;
  'task:updated': (data: { taskId: string; status: string; progress?: number }) => void;
  'task:completed': (data: { taskId: string; status: string; result?: any }) => void;
  'task:failed': (data: { taskId: string; status: string; error: string }) => void;
  
  // System events
  'system:status': (data: { component: string; status: string; message?: string }) => void;
  'system:metrics': (data: { cpu: number; memory: number; disk: number }) => void;
  
  // File events
  'file:uploaded': (data: { fileId: string; filename: string; size: number }) => void;
  'file:processed': (data: { fileId: string; taskId: string; status: string }) => void;
  
  // Connection events
  connect: () => void;
  disconnect: () => void;
  connect_error: (error: Error) => void;
}

interface SocketContextType {
  socket: Socket | null;
  isConnected: boolean;
  subscribe: <K extends keyof SocketEvents>(event: K, callback: SocketEvents[K]) => void;
  unsubscribe: <K extends keyof SocketEvents>(event: K, callback?: SocketEvents[K]) => void;
  emit: (event: string, data?: any) => void;
}

const SocketContext = createContext<SocketContextType>({
  socket: null,
  isConnected: false,
  subscribe: () => {},
  unsubscribe: () => {},
  emit: () => {},
});

interface SocketProviderProps {
  children: ReactNode;
}

export const SocketProvider: React.FC<SocketProviderProps> = ({ children }) => {
  const [socket, setSocket] = useState<Socket | null>(null);
  const [isConnected, setIsConnected] = useState(false);

  useEffect(() => {
    // Initialize socket connection
    const socketUrl = process.env.REACT_APP_SOCKET_URL || 'http://localhost:8000';
    
    const newSocket = io(socketUrl, {
      transports: ['websocket', 'polling'],
      timeout: 20000,
      reconnection: true,
      reconnectionAttempts: 5,
      reconnectionDelay: 1000,
      auth: {
        token: localStorage.getItem('auth_token'),
      },
    });

    // Connection event handlers
    newSocket.on('connect', () => {
      console.log('Socket connected:', newSocket.id);
      setIsConnected(true);
      toast.success('Connected to server', { id: 'socket-connection' });
    });

    newSocket.on('disconnect', (reason) => {
      console.log('Socket disconnected:', reason);
      setIsConnected(false);
      
      if (reason === 'io server disconnect') {
        // Server disconnected - try to reconnect
        newSocket.connect();
      }
      
      toast.error('Disconnected from server', { id: 'socket-connection' });
    });

    newSocket.on('connect_error', (error) => {
      console.error('Socket connection error:', error);
      setIsConnected(false);
      toast.error('Connection failed', { id: 'socket-connection' });
    });

    newSocket.on('reconnect', (attemptNumber) => {
      console.log('Socket reconnected after', attemptNumber, 'attempts');
      toast.success('Reconnected to server', { id: 'socket-connection' });
    });

    newSocket.on('reconnect_error', (error) => {
      console.error('Socket reconnection error:', error);
    });

    newSocket.on('reconnect_failed', () => {
      console.error('Socket reconnection failed');
      toast.error('Failed to reconnect to server', { id: 'socket-connection' });
    });

    // Global event handlers for notifications
    newSocket.on('task:created', (data) => {
      toast.success(`Task created: ${data.filename}`, {
        duration: 3000,
        icon: '📁',
      });
    });

    newSocket.on('task:completed', (data) => {
      toast.success(`Task completed: ${data.taskId}`, {
        duration: 5000,
        icon: '✅',
      });
    });

    newSocket.on('task:failed', (data) => {
      toast.error(`Task failed: ${data.error}`, {
        duration: 5000,
        icon: '❌',
      });
    });

    newSocket.on('system:status', (data) => {
      if (data.status === 'error') {
        toast.error(`System issue: ${data.component} - ${data.message}`, {
          duration: 10000,
          icon: '⚠️',
        });
      }
    });

    setSocket(newSocket);

    // Cleanup on unmount
    return () => {
      newSocket.close();
    };
  }, []);

  const subscribe = <K extends keyof SocketEvents>(
    event: K,
    callback: SocketEvents[K]
  ) => {
    if (socket) {
      socket.on(event as string, callback as any);
    }
  };

  const unsubscribe = <K extends keyof SocketEvents>(
    event: K,
    callback?: SocketEvents[K]
  ) => {
    if (socket) {
      if (callback) {
        socket.off(event as string, callback as any);
      } else {
        socket.removeAllListeners(event as string);
      }
    }
  };

  const emit = (event: string, data?: any) => {
    if (socket && isConnected) {
      socket.emit(event, data);
    } else {
      console.warn('Socket not connected, cannot emit event:', event);
    }
  };

  const value: SocketContextType = {
    socket,
    isConnected,
    subscribe,
    unsubscribe,
    emit,
  };

  return (
    <SocketContext.Provider value={value}>
      {children}
    </SocketContext.Provider>
  );
};

// Custom hook to use socket
export const useSocket = () => {
  const context = useContext(SocketContext);
  
  if (!context) {
    throw new Error('useSocket must be used within a SocketProvider');
  }
  
  return context;
};

// Custom hook for task updates
export const useTaskUpdates = (taskId?: string) => {
  const { subscribe, unsubscribe } = useSocket();
  const [taskStatus, setTaskStatus] = useState<{
    status: string;
    progress?: number;
    error?: string;
  } | null>(null);

  useEffect(() => {
    if (!taskId) return;

    const handleTaskUpdate = (data: { taskId: string; status: string; progress?: number }) => {
      if (data.taskId === taskId) {
        setTaskStatus({
          status: data.status,
          progress: data.progress,
        });
      }
    };

    const handleTaskCompleted = (data: { taskId: string; status: string; result?: any }) => {
      if (data.taskId === taskId) {
        setTaskStatus({
          status: data.status,
          progress: 100,
        });
      }
    };

    const handleTaskFailed = (data: { taskId: string; status: string; error: string }) => {
      if (data.taskId === taskId) {
        setTaskStatus({
          status: data.status,
          error: data.error,
        });
      }
    };

    subscribe('task:updated', handleTaskUpdate);
    subscribe('task:completed', handleTaskCompleted);
    subscribe('task:failed', handleTaskFailed);

    return () => {
      unsubscribe('task:updated', handleTaskUpdate);
      unsubscribe('task:completed', handleTaskCompleted);
      unsubscribe('task:failed', handleTaskFailed);
    };
  }, [taskId, subscribe, unsubscribe]);

  return taskStatus;
};

// Custom hook for system status
export const useSystemStatus = () => {
  const { subscribe, unsubscribe } = useSocket();
  const [systemStatus, setSystemStatus] = useState<{
    components: Record<string, { status: string; message?: string }>;
    metrics?: { cpu: number; memory: number; disk: number };
  }>({
    components: {},
  });

  useEffect(() => {
    const handleSystemStatus = (data: { component: string; status: string; message?: string }) => {
      setSystemStatus(prev => ({
        ...prev,
        components: {
          ...prev.components,
          [data.component]: {
            status: data.status,
            message: data.message,
          },
        },
      }));
    };

    const handleSystemMetrics = (data: { cpu: number; memory: number; disk: number }) => {
      setSystemStatus(prev => ({
        ...prev,
        metrics: data,
      }));
    };

    subscribe('system:status', handleSystemStatus);
    subscribe('system:metrics', handleSystemMetrics);

    return () => {
      unsubscribe('system:status', handleSystemStatus);
      unsubscribe('system:metrics', handleSystemMetrics);
    };
  }, [subscribe, unsubscribe]);

  return systemStatus;
};

export default SocketContext;
