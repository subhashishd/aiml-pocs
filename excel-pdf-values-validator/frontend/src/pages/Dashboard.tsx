import React from 'react';
import { useQuery } from 'react-query';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import {
  FiUpload,
  FiFileText,
  FiCheckCircle,
  FiAlertCircle,
  FiClock,
  FiActivity,
  FiTrendingUp,
  FiArrowRight,
} from 'react-icons/fi';

import { Card } from '../components/ui/Card';
import { Button } from '../components/ui/Button';
import { Badge } from '../components/ui/Badge';
import { LoadingSpinner } from '../components/ui/LoadingSpinner';
import { api } from '../services/apiClient';

const DashboardContainer = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

const WelcomeSection = styled.div`
  margin-bottom: 2rem;
`;

const WelcomeTitle = styled.h1`
  color: ${({ theme }) => theme.colors.gray[900]};
  font-size: 2rem;
  font-weight: 700;
  margin: 0 0 0.5rem 0;
`;

const WelcomeSubtitle = styled.p`
  color: ${({ theme }) => theme.colors.gray[600]};
  font-size: 1.1rem;
  margin: 0;
`;

const StatsGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 1.5rem;
  margin-bottom: 2rem;
`;

const StatCard = styled(Card)`
  display: flex;
  align-items: center;
  padding: 1.5rem;
`;

const StatIcon = styled.div<{ color: string }>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 48px;
  height: 48px;
  border-radius: 12px;
  background-color: ${({ color }) => color};
  margin-right: 1rem;

  svg {
    color: white;
  }
`;

const StatContent = styled.div`
  flex: 1;
`;

const StatValue = styled.div`
  font-size: 2rem;
  font-weight: 700;
  color: ${({ theme }) => theme.colors.gray[900]};
  line-height: 1;
  margin-bottom: 0.25rem;
`;

const StatLabel = styled.div`
  font-size: 0.875rem;
  color: ${({ theme }) => theme.colors.gray[600]};
`;

const ContentGrid = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  gap: 2rem;

  @media (max-width: 1024px) {
    grid-template-columns: 1fr;
  }
`;

const RecentTasksSection = styled.div``;

const QuickActionsSection = styled.div``;

const SectionHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 1rem;
`;

const SectionTitle = styled.h2`
  color: ${({ theme }) => theme.colors.gray[900]};
  font-size: 1.25rem;
  font-weight: 600;
  margin: 0;
`;

const ViewAllLink = styled(Link)`
  color: ${({ theme }) => theme.colors.primary[600]};
  text-decoration: none;
  font-size: 0.875rem;
  font-weight: 500;
  display: flex;
  align-items: center;
  gap: 0.25rem;

  &:hover {
    color: ${({ theme }) => theme.colors.primary[700]};
  }
`;

const TaskList = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.75rem;
`;

const TaskItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
  background-color: ${({ theme }) => theme.colors.white};
  border: 1px solid ${({ theme }) => theme.colors.gray[200]};
  border-radius: 8px;
  transition: all 0.2s ease;

  &:hover {
    border-color: ${({ theme }) => theme.colors.gray[300]};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
  }
`;

const TaskInfo = styled.div`
  flex: 1;
`;

const TaskTitle = styled.div`
  font-weight: 500;
  color: ${({ theme }) => theme.colors.gray[900]};
  margin-bottom: 0.25rem;
`;

const TaskMeta = styled.div`
  font-size: 0.875rem;
  color: ${({ theme }) => theme.colors.gray[600]};
`;

const TaskActions = styled.div`
  display: flex;
  align-items: center;
  gap: 0.75rem;
`;

const QuickActionGrid = styled.div`
  display: grid;
  gap: 1rem;
`;

const QuickActionCard = styled(Card)`
  padding: 1.5rem;
  text-align: center;
  cursor: pointer;
  transition: all 0.2s ease;
  border: 2px solid transparent;

  &:hover {
    border-color: ${({ theme }) => theme.colors.primary[200]};
    transform: translateY(-2px);
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
  }
`;

const QuickActionIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 64px;
  height: 64px;
  border-radius: 16px;
  background-color: ${({ theme }) => theme.colors.primary[100]};
  margin: 0 auto 1rem;

  svg {
    color: ${({ theme }) => theme.colors.primary[600]};
  }
`;

const QuickActionTitle = styled.h3`
  color: ${({ theme }) => theme.colors.gray[900]};
  font-size: 1.125rem;
  font-weight: 600;
  margin: 0 0 0.5rem 0;
`;

const QuickActionDescription = styled.p`
  color: ${({ theme }) => theme.colors.gray[600]};
  font-size: 0.875rem;
  margin: 0;
`;

const EmptyState = styled.div`
  text-align: center;
  padding: 2rem;
  color: ${({ theme }) => theme.colors.gray[500]};
`;

interface DashboardStats {
  totalTasks: number;
  completedTasks: number;
  failedTasks: number;
  pendingTasks: number;
}

interface Task {
  id: string;
  filename: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  created_at: string;
  file_type: 'pdf' | 'excel';
}

const getStatusBadgeProps = (status: string) => {
  switch (status) {
    case 'completed':
      return { variant: 'success' as const, icon: FiCheckCircle };
    case 'failed':
      return { variant: 'error' as const, icon: FiAlertCircle };
    case 'processing':
      return { variant: 'warning' as const, icon: FiClock };
    default:
      return { variant: 'default' as const, icon: FiClock };
  }
};

const Dashboard: React.FC = () => {
  const { data: stats, isLoading: statsLoading } = useQuery<DashboardStats>(
    'dashboard-stats',
    () => api.get('/dashboard/stats'),
    {
      refetchInterval: 30000, // Refresh every 30 seconds
    }
  );

  const { data: recentTasks, isLoading: tasksLoading } = useQuery<Task[]>(
    'recent-tasks',
    () => api.get('/tasks/recent?limit=5'),
    {
      refetchInterval: 30000,
    }
  );

  const formatDate = (dateString: string) => {
    return new Date(dateString).toLocaleDateString('en-US', {
      month: 'short',
      day: 'numeric',
      hour: '2-digit',
      minute: '2-digit',
    });
  };

  return (
    <DashboardContainer>
      <WelcomeSection>
        <WelcomeTitle>Welcome to ValidatorAI</WelcomeTitle>
        <WelcomeSubtitle>
          Upload and validate your PDF and Excel files with AI-powered intelligence
        </WelcomeSubtitle>
      </WelcomeSection>

      <StatsGrid>
        <StatCard>
          <StatIcon color="#3b82f6">
            <FiFileText size={24} />
          </StatIcon>
          <StatContent>
            <StatValue>{statsLoading ? '--' : stats?.totalTasks || 0}</StatValue>
            <StatLabel>Total Tasks</StatLabel>
          </StatContent>
        </StatCard>

        <StatCard>
          <StatIcon color="#10b981">
            <FiCheckCircle size={24} />
          </StatIcon>
          <StatContent>
            <StatValue>{statsLoading ? '--' : stats?.completedTasks || 0}</StatValue>
            <StatLabel>Completed</StatLabel>
          </StatContent>
        </StatCard>

        <StatCard>
          <StatIcon color="#f59e0b">
            <FiClock size={24} />
          </StatIcon>
          <StatContent>
            <StatValue>{statsLoading ? '--' : stats?.pendingTasks || 0}</StatValue>
            <StatLabel>In Progress</StatLabel>
          </StatContent>
        </StatCard>

        <StatCard>
          <StatIcon color="#ef4444">
            <FiAlertCircle size={24} />
          </StatIcon>
          <StatContent>
            <StatValue>{statsLoading ? '--' : stats?.failedTasks || 0}</StatValue>
            <StatLabel>Failed</StatLabel>
          </StatContent>
        </StatCard>
      </StatsGrid>

      <ContentGrid>
        <RecentTasksSection>
          <Card>
            <SectionHeader>
              <SectionTitle>Recent Tasks</SectionTitle>
              <ViewAllLink to="/results">
                View All <FiArrowRight size={14} />
              </ViewAllLink>
            </SectionHeader>

            {tasksLoading ? (
              <LoadingSpinner />
            ) : recentTasks && recentTasks.length > 0 ? (
              <TaskList>
                {recentTasks.map((task) => {
                  const { variant, icon: StatusIcon } = getStatusBadgeProps(task.status);
                  return (
                    <TaskItem key={task.id}>
                      <TaskInfo>
                        <TaskTitle>{task.filename}</TaskTitle>
                        <TaskMeta>{formatDate(task.created_at)}</TaskMeta>
                      </TaskInfo>
                      <TaskActions>
                        <Badge variant={variant}>
                          <StatusIcon size={14} />
                          {task.status}
                        </Badge>
                        <Button
                          as={Link}
                          to={`/results/${task.id}`}
                          variant="ghost"
                          size="small"
                        >
                          View
                        </Button>
                      </TaskActions>
                    </TaskItem>
                  );
                })}
              </TaskList>
            ) : (
              <EmptyState>
                <FiFileText size={48} style={{ opacity: 0.3, marginBottom: '1rem' }} />
                <p>No tasks yet. Upload your first file to get started!</p>
              </EmptyState>
            )}
          </Card>
        </RecentTasksSection>

        <QuickActionsSection>
          <Card>
            <SectionHeader>
              <SectionTitle>Quick Actions</SectionTitle>
            </SectionHeader>

            <QuickActionGrid>
              <QuickActionCard as={Link} to="/upload">
                <QuickActionIcon>
                  <FiUpload size={32} />
                </QuickActionIcon>
                <QuickActionTitle>Upload Files</QuickActionTitle>
                <QuickActionDescription>
                  Upload PDF or Excel files for validation
                </QuickActionDescription>
              </QuickActionCard>

              <QuickActionCard as={Link} to="/system">
                <QuickActionIcon>
                  <FiActivity size={32} />
                </QuickActionIcon>
                <QuickActionTitle>System Status</QuickActionTitle>
                <QuickActionDescription>
                  Check system health and performance
                </QuickActionDescription>
              </QuickActionCard>

              <QuickActionCard as={Link} to="/monitoring">
                <QuickActionIcon>
                  <FiTrendingUp size={32} />
                </QuickActionIcon>
                <QuickActionTitle>Monitoring</QuickActionTitle>
                <QuickActionDescription>
                  View real-time system metrics
                </QuickActionDescription>
              </QuickActionCard>
            </QuickActionGrid>
          </Card>
        </QuickActionsSection>
      </ContentGrid>
    </DashboardContainer>
  );
};

export default Dashboard;
