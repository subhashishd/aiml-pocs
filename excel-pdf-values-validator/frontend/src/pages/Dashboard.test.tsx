import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import { render, createTestQueryClient } from '../test-utils';
import Dashboard from './Dashboard';
import { api } from '../services/apiClient';

// Mock the getStats and getRecent API calls
jest.mock('../services/apiClient', () => ({
  api: {
    get: jest.fn(),
  },
}));

const mockedApi = api as jest.Mocked<typeof api>;

// Sample test data
const statsMockData = {
  totalTasks: 10,
  completedTasks: 7,
  failedTasks: 1,
  pendingTasks: 2,
};

const recentTasksMockData = [
  { id: '1', filename: 'file1.pdf', status: 'completed', created_at: '2023-01-01T00:00:00Z', file_type: 'pdf' },
  { id: '2', filename: 'file2.pdf', status: 'failed', created_at: '2023-01-02T00:00:00Z', file_type: 'pdf' },
  { id: '3', filename: 'file3.pdf', status: 'processing', created_at: '2023-01-03T00:00:00Z', file_type: 'pdf' },
];

// Test suite for Dashboard
describe('Dashboard Page', () => {
  beforeEach(() => {
    // Clear all mocks before each test
    jest.clearAllMocks();
    mockedApi.get.mockClear();
  });

  // Test for successful data fetching
  it('renders the dashboard with data', async () => {
    // Mock the API response
    mockedApi.get
      .mockResolvedValueOnce(statsMockData)
      .mockResolvedValueOnce(recentTasksMockData);

    const queryClient = createTestQueryClient();

    // Render the component
    render(<Dashboard />, { queryClient });

    // Wait for element to be in the document
    await waitFor(() => expect(screen.getByText('Welcome to ValidatorAI')).toBeInTheDocument());

    // Wait for data to load and check for the rendered elements
    await waitFor(() => {
      expect(screen.getByText('10')).toBeInTheDocument();
    }, { timeout: 5000 });
    expect(screen.getByText('Completed')).toBeInTheDocument();
    expect(screen.getByText('file1.pdf')).toBeInTheDocument();
    expect(screen.getByText('file2.pdf')).toBeInTheDocument();
  });

  // Test for loading state
  it('shows loading state initially', () => {
    // Mock API to never resolve (simulate loading)
    mockedApi.get.mockImplementation(() => new Promise(() => {}));
    
    const queryClient = createTestQueryClient();

    // Render the component
    render(<Dashboard />, { queryClient });

    // Should show loading indicators
    expect(screen.getByText('Welcome to ValidatorAI')).toBeInTheDocument();
    expect(screen.getAllByText('--')).toHaveLength(4); // All stat cards show "--" during loading
  });

  // Test for error state - component doesn't show error message, just fails silently
  it('handles error state when data fetching fails', async () => {
    // Mock the API error response with proper error silencing
    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    
    mockedApi.get.mockRejectedValue(new Error('Server error'));

    const queryClient = createTestQueryClient();

    // Render the component
    render(<Dashboard />, { queryClient });

    // Should still show the basic dashboard layout
    expect(screen.getByText('Welcome to ValidatorAI')).toBeInTheDocument();
    expect(screen.getByText('Total Tasks')).toBeInTheDocument();
    
    // Wait for error state to settle
    await waitFor(() => {
      expect(screen.getByText('No tasks yet. Upload your first file to get started!')).toBeInTheDocument();
    }, { timeout: 3000 });
    
    consoleErrorSpy.mockRestore();
  });

  it('matches snapshot', () => {
    const queryClient = createTestQueryClient();

    // Render the component
    const { asFragment } = render(<Dashboard />, { queryClient });

    // Snapshot match
    expect(asFragment()).toMatchSnapshot();
  });

});
