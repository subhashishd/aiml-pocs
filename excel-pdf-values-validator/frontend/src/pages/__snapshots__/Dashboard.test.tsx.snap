// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="mock-socket-provider"
  >
    <div
      class="sc-dhKdcB kZRLJN"
    >
      <div
        class="sc-kpDqfm laYjnQ"
      >
        <h1
          class="sc-dAlyuH elwjJr"
        >
          Welcome to ValidatorAI
        </h1>
        <p
          class="sc-jlZhew eZFvko"
        >
          Upload and validate your PDF and Excel files with AI-powered intelligence
        </p>
      </div>
      <div
        class="sc-cwHptR iTKmhT"
      >
        <div
          class="sc-aXZVg sc-jEACwC egUycH bNvlxB"
        >
          <div
            class="sc-cPiKLX fGWVdn"
            color="#3b82f6"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
              />
              <polyline
                points="14 2 14 8 20 8"
              />
              <line
                x1="16"
                x2="8"
                y1="13"
                y2="13"
              />
              <line
                x1="16"
                x2="8"
                y1="17"
                y2="17"
              />
              <polyline
                points="10 9 9 9 8 9"
              />
            </svg>
          </div>
          <div
            class="sc-dLMFU kyCAfB"
          >
            <div
              class="sc-eDPEul bsyygd"
            >
              --
            </div>
            <div
              class="sc-eldPxv lorvlq"
            >
              Total Tasks
            </div>
          </div>
        </div>
        <div
          class="sc-aXZVg sc-jEACwC egUycH bNvlxB"
        >
          <div
            class="sc-cPiKLX HOYZf"
            color="#10b981"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
          </div>
          <div
            class="sc-dLMFU kyCAfB"
          >
            <div
              class="sc-eDPEul bsyygd"
            >
              --
            </div>
            <div
              class="sc-eldPxv lorvlq"
            >
              Completed
            </div>
          </div>
        </div>
        <div
          class="sc-aXZVg sc-jEACwC egUycH bNvlxB"
        >
          <div
            class="sc-cPiKLX edZjZJ"
            color="#f59e0b"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
          </div>
          <div
            class="sc-dLMFU kyCAfB"
          >
            <div
              class="sc-eDPEul bsyygd"
            >
              --
            </div>
            <div
              class="sc-eldPxv lorvlq"
            >
              In Progress
            </div>
          </div>
        </div>
        <div
          class="sc-aXZVg sc-jEACwC egUycH bNvlxB"
        >
          <div
            class="sc-cPiKLX beQzPT"
            color="#ef4444"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="16"
                y2="16"
              />
            </svg>
          </div>
          <div
            class="sc-dLMFU kyCAfB"
          >
            <div
              class="sc-eDPEul bsyygd"
            >
              --
            </div>
            <div
              class="sc-eldPxv lorvlq"
            >
              Failed
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-fPXMVe hdVhJb"
      >
        <div
          class="sc-gFqAkR"
        >
          <div
            class="sc-aXZVg egUycH"
          >
            <div
              class="sc-dAbbOL cDQopq"
            >
              <h2
                class="sc-feUZmu krljLk"
              >
                Recent Tasks
              </h2>
              <a
                class="sc-fUnMCh cTzgOt"
                href="/results"
              >
                View All 
                <svg
                  fill="none"
                  height="14"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                  <polyline
                    points="12 5 19 12 12 19"
                  />
                </svg>
              </a>
            </div>
            <div
              class="sc-imWYAI gsWVGT"
            >
              <div
                class="sc-jXbUNg faFWmQ"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-ikkxIA"
        >
          <div
            class="sc-aXZVg egUycH"
          >
            <div
              class="sc-dAbbOL cDQopq"
            >
              <h2
                class="sc-feUZmu krljLk"
              >
                Quick Actions
              </h2>
            </div>
            <div
              class="sc-koXPp fEuCmx"
            >
              <a
                class="sc-aXZVg sc-bmzYkS egUycH ixZckJ"
                href="/upload"
              >
                <div
                  class="sc-iHGNWf kufNkN"
                >
                  <svg
                    fill="none"
                    height="32"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
                    />
                    <polyline
                      points="17 8 12 3 7 8"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="3"
                      y2="15"
                    />
                  </svg>
                </div>
                <h3
                  class="sc-dtBdUo YHMMo"
                >
                  Upload Files
                </h3>
                <p
                  class="sc-kOHTFB hpktpA"
                >
                  Upload PDF or Excel files for validation
                </p>
              </a>
              <a
                class="sc-aXZVg sc-bmzYkS egUycH ixZckJ"
                href="/system"
              >
                <div
                  class="sc-iHGNWf kufNkN"
                >
                  <svg
                    fill="none"
                    height="32"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="22 12 18 12 15 21 9 3 6 12 2 12"
                    />
                  </svg>
                </div>
                <h3
                  class="sc-dtBdUo YHMMo"
                >
                  System Status
                </h3>
                <p
                  class="sc-kOHTFB hpktpA"
                >
                  Check system health and performance
                </p>
              </a>
              <a
                class="sc-aXZVg sc-bmzYkS egUycH ixZckJ"
                href="/monitoring"
              >
                <div
                  class="sc-iHGNWf kufNkN"
                >
                  <svg
                    fill="none"
                    height="32"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="23 6 13.5 15.5 8.5 10.5 1 18"
                    />
                    <polyline
                      points="17 6 23 6 23 12"
                    />
                  </svg>
                </div>
                <h3
                  class="sc-dtBdUo YHMMo"
                >
                  Monitoring
                </h3>
                <p
                  class="sc-kOHTFB hpktpA"
                >
                  View real-time system metrics
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
